gc/base/MemorySubSpaceGenerational.cpp

```diff
@@ -325,7 +325,20 @@ MM_MemorySubSpaceGenerational::counterBalanceContract(
 uintptr_t
 MM_MemorySubSpaceGenerational::releaseFreeMemoryPages(MM_EnvironmentBase* env)
 {
-	return _memorySubSpaceOld->releaseFreeMemoryPages(env);
+	return releaseFreeMemoryPages(env, MEMORY_TYPE_OLD);
+}
+
+uintptr_t
+MM_MemorySubSpaceGenerational::releaseFreeMemoryPages(MM_EnvironmentBase* env, uintptr_t includeMemoryType)
+{
+	uintptr_t releasedPages = 0;
+	if (OMR_ARE_ALL_BITS_SET(includeMemoryType, MEMORY_TYPE_OLD)) {
+		releasedPages += _memorySubSpaceOld->releaseFreeMemoryPages(env, MEMORY_TYPE_OLD);
+	}
+	if (OMR_ARE_ALL_BITS_SET(includeMemoryType, MEMORY_TYPE_NEW)) {
+		releasedPages += _memorySubSpaceNew->releaseFreeMemoryPages(env, MEMORY_TYPE_NEW);
+	}
```

Comment:
This is the only place where we really need (for decision making) to pass and interpret memory type. Anywhere else, if needed at all, is to satisfy compiler. Let me think if we can simplify the rest of the code...

Reply:
We should call just default (no arguments) releaseFreeMemoryPages here